.devcontainer/devcontainer.json

```diff
@@ -1,39 +1,31 @@
+// For more details, see https://aka.ms/devcontainer.json.
 {
-  "name": "Mastodon",
-  "dockerComposeFile": "docker-compose.yml",
-  "service": "app",
-  "workspaceFolder": "/mastodon",
+		"name": "Mastodon",
+		"dockerComposeFile": "docker-compose.yml",
+		"service": "app",
+		"workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}",
 
-  // Configure tool-specific properties.
-  "customizations": {
-    // Configure properties specific to VS Code.
-    "vscode": {
-      // Set *default* container specific settings.json values on container create.
-      "settings": {},
+		// Features to add to the dev container. More info: https://containers.dev/features.
+		"features": {
+				"ghcr.io/devcontainers/features/sshd:1": {}
+		},
 
-      // Add the IDs of extensions you want installed when the container is created.
-      "extensions": [
-        "EditorConfig.EditorConfig",
-        "dbaeumer.vscode-eslint",
-        "rebornix.Ruby",
-        "webben.browserslist"
-      ]
-    }
-  },
+		// Use 'forwardPorts' to make a list of ports inside the container available locally.
+		// This can be used to network with other containers or the host.
+		"forwardPorts": [3000, 4000],
 
-  "features": {
-    "ghcr.io/devcontainers/features/sshd:1": {
-      "version": "latest"
-    }
-  },
+		// Use 'postCreateCommand' to run commands after the container is created.
+		"postCreateCommand": ".devcontainer/post-create.sh",
+		"waitFor": "postCreateCommand",
 
-  // Use 'forwardPorts' to make a list of ports inside the container available locally.
-  // This can be used to network with other containers or the host.
-  "forwardPorts": [3000, 4000],
-
-  // Use 'postCreateCommand' to run commands after the container is created.
-  "postCreateCommand": ".devcontainer/post-create.sh",
-
-  // Comment out to connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
-  "remoteUser": "vscode"
+		// Configure tool-specific properties.
+		"customizations": {
+				// Configure properties specific to VS Code.
+				"vscode": {
+						// Set *default* container specific settings.json values on container create.
+						"settings": {},
+						// Add the IDs of extensions you want installed when the container is created.
+						"extensions": ["EditorConfig.EditorConfig", "webben.browserslist"]
```

Comment:
Why remove the ruby and eslint extensions?

Reply:
I am trying to cleanup duplication, the ruby image used by this dev container already adds the [rebornix.Ruby here](https://github.com/devcontainers/images/blob/main/src/ruby/.devcontainer/devcontainer.json#L27). Also, it adds the Node Feature which adds the [eslint extension here](https://github.com/devcontainers/features/blob/main/src/node/devcontainer-feature.json#L40). 

The supporting tools using the dev container CLI [metadata logic](https://containers.dev/implementors/spec/#merge-logic) merges these customizations.